hack/grab-profiles.sh

```diff
@@ -0,0 +1,175 @@
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+server_addr=""
+requested_profiles=""
+mem_pprof_flags=""
+profile_components=""
+output_dir="."
+tunnel_port="${tunnel_port:-1234}"
+
+args=$(getopt -o s:mho: -l server:,master,heapster,output:,help,inuse-space,inuse-objects,alloc-space,alloc-objects,cpu -- "$@")
+if [[ $? -ne 0 ]]; then
+  >&2 echo "Error in getopt"
+  exit 1
+fi
+
+HEAPSTER_VERSION="v0.18.2"
+MASTER_PPROF_PATH="debug/pprof"
+HEAPSTER_PPROF_PATH="api/v1/proxy/namespaces/kube-system/services/monitoring-heapster/debug/pprof"
+
+eval set -- "${args}"
+
+while true; do
+  case $1 in
+    -s|--server)
+      shift
+      if [ -z "$1" ]; then
+        >&2 echo "empty argument to --server flag"
+        exit 1
+      fi
+      server_addr=$1
+      shift
+      ;;
+    -m|--master)
+      shift
+      profile_components="master ${profile_components}"
+      ;;
+    -h|--heapster)
+      shift
+      profile_components="heapster ${profile_components}"
+      ;;
+    -o|--output)
+      shift
+      if [ -z "$1" ]; then
+        >&2 echo "empty argument to --output flag"
+        exit 1
+      fi
+      output_dir=$1
+      shift
+      ;;
+    --inuse-space)
+      shift
+      requested_profiles="mem ${requested_profiles}"
+      mem_pprof_flags="inuse_space ${mem_pprof_flags}"
+      ;;
+    --inuse-objects)
+      shift
+      requested_profiles="mem ${requested_profiles}"
+      mem_pprof_flags="inuse_objects ${mem_pprof_flags}"
+      ;;
+    --alloc-space)
+      shift
+      requested_profiles="mem ${requested_profiles}"
+      mem_pprof_flags="alloc_space ${mem_pprof_flags}"
+      ;;
+    --alloc-objects)
+      shift
+      requested_profiles="mem ${requested_profiles}"
+      mem_pprof_flags="alloc_objects ${mem_pprof_flags}"
+      ;;
+    --cpu)
+      shift
+      requested_profiles="cpu"
+      ;;
+    --help)
+      shift
+      echo "Recognized options:
+        -o/--output,
+        -s/--server,
+        -m/--master,
+        -h/--heapster,
+        --inuse-space,
+        --inuse-objects,
+        --alloc-space,
+        --alloc-objects,
+        --cpu,
+        --help"
+      exit 0
+      ;;
+    --)
+      shift
+      break;
+      ;;
+  esac
+done
+
+if [[ -z "${server_addr}" ]]; then
+  >&2 echo "Server flag is required"
+  exit 1
+fi
+
+if [[ -z "${profile_components}" ]]; then
+  >&2 echo "Choose at least one component to profile"
+  exit 1
+fi
+
+if [[ -z "${requested_profiles}" ]]; then
+  >&2 echo "Choose at least one profiling option"
+  exit 1
+fi
+
+gcloud compute ssh "${server_addr}" --ssh-flag=-nN --ssh-flag=-L${tunnel_port}:localhost:8080 &
+
+echo "Waiting for tunnel to be created..."
+kube::util::wait_for_url http://localhost:1234/healthz
+
+SSH_PID=$(ps aux | grep "1234:localhost:8080" | grep ssh | tr -s " " | cut -f2 -d" ")
```

Comment:
should use ${tunnel_port} instead of 1234.

also, pgrep may be cleaner, e.g. `pgrep -f "ssh.*${tunnel_port}:localhost:8080"`